model_analyzer/entrypoint.py

```diff
@@ -121,6 +119,21 @@ def get_server_handle(config):
             log_path=config.triton_output_path,
             mounts=config.triton_docker_mounts,
             labels=config.triton_docker_labels)
+    elif config.triton_launch_mode == 'C_API':
+        triton_config = TritonServerConfig()
+        triton_config['model-repository'] = os.path.abspath(
+            config.output_model_repository_path)
+        logger.info("Starting a Triton Server using perf_analyzer's C_API...")
+        server = TritonServerFactory.create_server_local(path=None,
+                                                         config=triton_config,
+                                                         gpus=[],
+                                                         log_path="")
+        logger.warning(
+            "Model Analyzer currently does not support setting custom Triton server "
+            "flags when profiling with perf_analyzer's C_API. Additionally, some "
+            "metrics maybe affected as Triton is not launched with explicit model "
+            "control mode, and as a result, loads all model config variants as they "
```

Comment:
I wonder how the C API works if it is going to run all the models in the output model repository? Some of the models in the output model repository only contain config file and they are not loadable without the model files.

Reply:
I think without Katherine's changes it is not possible to run config search in C-API.

Reply:
I changed the model repository management to use symlinks. You can check https://github.com/triton-inference-server/model_analyzer/blob/2110347c7fae9f07ead6c5972eda45d9b3a48134/model_analyzer/triton/model/model_config.py#L197

Reply:
Oh, I see. Thanks for the link.